src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1387,9 +1386,10 @@ private static void ThrowTimeoutTooLarge(TimeSpan? timeout)
             throw new InvalidOperationException($"Maximum request timeout is set to '{GlobalHttpClientTimeout}' but was '{timeout}'.");
         }
 
-        private HttpCache.ReleaseCacheItem GetFromCache<TResult>(JsonOperationContext context, RavenCommand<TResult> command, bool useCache, string url, out string cachedChangeVector, out BlittableJsonReaderObject cachedValue)
+        private HttpCache.ReleaseCacheItem GetFromCache<TResult>(JsonOperationContext context, RavenCommand<TResult> command, SessionInfo sessionInfo, string url, out string cachedChangeVector, out BlittableJsonReaderObject cachedValue)
         {
-            if (useCache && command.CanCache && command.IsReadRequest && command.ResponseType == RavenCommandResponseType.Object)
+            command.NoCaching = sessionInfo?.NoCaching ?? false;
```

Comment:
isn't this a matter of doing:
```csharp
if (sessionInfo?.NoCaching == true)
    command.CanCache = false;
```

instead of introducing a new property?

Reply:
The CanCache property has a protected access modifier, so I assumed it should not be modified directly. 
Would it be acceptable to change its access level to internal for this adjustment?

Reply:
Try `protected internal`

Reply:
Updated as suggested. Thank you for the guidance!